src/array.js

```diff
@@ -22,14 +26,15 @@ if (!Array.from) {
         , C = typeof this == 'function' ? this : Array
         , aLen = arguments.length
         , mapfn = aLen > 1 ? arguments[1] : undefined
+        , thisArg = aLen > 2 ? arguments[2] : undefined
         , mapping = mapfn !== undefined
         , index = 0
         , iterFn = O[Symbol.iterator]
         , length, result, step, iterator;
       if (mapping) mapfn = mapfn.bind(aLen > 2 ? arguments[2] : undefined);
```

Comment:
It seems this code was intended to perform the same function as your new code... 
So either your change was not required, or this can be removed.